dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/services/impl/JPAPersistenceManagerImpl.java

```diff
@@ -2407,7 +2414,13 @@ public boolean purgeJobInstanceAndRelatedData(final long jobInstanceId) {
             new TranRequest<Void>(em) {
                 @Override
                 public Void call() {
-                    final JobInstanceEntity instance = em.find(JobInstanceEntity.class, jobInstanceId);
+                    final JobInstanceEntity instance;
+
+                    if (instanceVersion == 3) {
+                        instance = em.find(JobInstanceEntityV3.class, jobInstanceId);
+                    } else
+                        instance = em.find(JobInstanceEntity.class, jobInstanceId);
+
```

Comment:
Don't think you need to do anything different for V3 here.

Reply:
extractor should give right instance version back so really should only need to do this:
instance = em.find(JobInstanceEntity.class, jobInstanceId);